### DISCUSSION TITLE
Query about the error "1 is not a supported code page. Parameter name: codepage" - SharpZipLib with 0.86 version.

### DISCUSSION THREAD

**Discussion by 30180536** [icsharpcode/SharpZipLib#857] [Q&A]
We have an issue with customer side using the application with 0.86 version of SharpZipLib. The reported error from application is "1 is not a supported code page. Parameter name: codepage". The regional settings in customer side is Dutch (Netherlands). The interesting thing is issue is happening only in customer side and not reproducible in our development environment.

Please let us know whether the issue is resolved in ZipLib latest version.

**Comment by piksel** [comment]
This is due to how the locales/codepages are handled in .NET. Try adding the following before the use of SharpZipLib:

```cs
Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```

Or, set another encoding by adding a `StringCodec` to the constructor (where applicable) or set the StringCodec property.

Ref: https://icsharpcode.github.io/SharpZipLib/help/api/ICSharpCode.SharpZipLib.Zip.StringCodec.html

Codepage `1` seems to be an invalid code, but it might carry some special meaning. Perhaps it's used due to .NET not recognizing the OS locale..?

**Reply by 30180536** [reply]
Thanks, piksel. Unfortunately, the application modification is not an immediate option now to implement the above-mentioned code.
The latest custom application having the SharpZipLib component with latest version (1.4.2). Does it resolve the issue with SharpZipLib  
version (1.4.2) ?

**Comment by SourceproStudio** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
